src/providers/log.provider.ts

```diff
@@ -0,0 +1,71 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: loopback4-example-log-extension
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {inject, Provider, Constructor} from '@loopback/context';
+import {CoreBindings} from '@loopback/core';
+import {OperationArgs, ParsedRequest} from '@loopback/rest';
+import {getLogMetadata} from '../decorators/log.decorator';
+import {EXAMPLE_LOG_BINDINGS, LOG_LEVEL} from '../keys';
+import {LogFn, Time, TimerFn} from '../types';
+
+export class LogActionProvider implements Provider<LogFn> {
+  constructor(
+    @inject(CoreBindings.CONTROLLER_CLASS)
+    private readonly controllerClass: Constructor<{}>,
+    @inject(CoreBindings.CONTROLLER_METHOD_NAME)
```

Comment:
Your code is trying to access current controller & method at the time the sequence action is instantiated, which happens at the beginning of request handling, before `findRoute` is called. The current controller & method are determined and bound to the context by `findRoute`, they are not available in the initial per-request context.

Read more here: http://loopback.io/doc/en/lb4/Creating-components.html#accessing-elements-contributed-by-other-sequence-actions